core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/geodistance/GeoDistanceParser.java

```diff
@@ -164,6 +162,11 @@ public AggregatorFactory parse(String aggregationName, XContentParser parser, Se
         return new GeoDistanceFactory(aggregationName, vsParser.config(), InternalGeoDistance.FACTORY, origin, unit, distanceType, ranges, keyed);
     }
 
+    private static String getSearchParseExceptionMessage(String currentFieldName, String aggregationName) {
+        return "Failed when parsing field: '" +  currentFieldName +
```

Comment:
other tokens are START_OBJECT, END_OBJECT, START_ARRAY, END_ARRAY, FIELD_NAME, VALUE_STRING, VALUE_NUMBER, VALUE_BOOLEAN, VALUE_EMBEDDED_OBJECT, VALUE_NULL ..

Most of the names simply don't make sense for me (who is also new to elastic search- so I am like layman now) ..

I personally don't like it for the simple reason that it is not understandable ....Unless user knows what each means it will not help.

Note: 
It is because of these problems I had the possible causes section which will take him straight to the possible fixes .... which will be generic ...